src/metabase/query_processor/pivot.clj

```diff
@@ -67,3 +72,87 @@
              (add-grouping-field request breakout
                                  (bitwise-not-truncate (count all-breakouts) start-mask))))
          new-breakouts)))
+
+(defn- process-query-append-results
+  "Reduce the results of a single `query` using `rf` and initial value `init`."
+  [query rf init context]
+  (if (a/poll! (qp.context/canceled-chan context))
+    (ensure-reduced init)
+    (qp/process-query-sync
+     query
+     {:canceled-chan (qp.context/canceled-chan context)
+      :rff           (fn [_]
+                       (fn
+                         ([]        init)
+                         ([acc]     acc)
+                         ([acc row] (rf acc ((:row-mapping-fn context) row context)))))})))
+
+(defn- process-queries-append-results
+  "Reduce the results of a sequence of `queries` using `rf` and initial value `init`."
+  [queries rf init context]
+  (reduce
+   (fn [acc query]
+     (process-query-append-results query rf acc (assoc context
+                                                       :pivot-column-mapping ((:column-mapping-fn context) query))))
+   init
+   queries))
+
+(defn- append-queries-context
+  "Update Query Processor `context` so it appends the rows fetched when running `more-queries`."
+  [context more-queries]
+  (cond-> context
+    (seq more-queries)
+    (update :rff (fn [rff]
+                   (fn [metadata]
+                     (let [rf (rff metadata)]
+                       (fn
+                         ([]        (rf))
+                         ([acc]     (rf (process-queries-append-results more-queries rf acc context)))
+                         ([acc row] (rf acc row)))))))))
+
+(defn process-multiple-queries
+  "Allows the query processor to handle multiple queries, stitched together to appear as one"
+  [[first-query & more-queries] context]
+  (qp/process-query first-query (append-queries-context context more-queries)))
+
+(defn run-pivot-query
+  "Run the pivot query. Unlike many query execution functions, this takes `context` as the first parameter to support
```

Comment:
while I have unhealthy love for `partial`, I'm not stoked about subtly changing argument order

Reply:
👍 Similar functions should have similar argument orders wherever possible. It makes our code more predictable. It increases cognitive load for people to have to remember `run-pivot-query` has the opposite arg order of basically all other QP functions... it's going to trip people up and slow REPL interaction speed down